### PR TITLE
@ #440 | more correct way to check branch

### PR DESCRIPTION
**Pull request by datphan**
to resolve some issues like branch name contains `\` or ref is at `detached HEAD`

connect #440

### CODE REVIEW

**Comment by hoatle** [comment]
@phuonglm @hieptranquoc please review this asap

**Review by hieptranquoc — APPROVED** [review]

File: lib/teracy-dev/location/git_synch.rb [hunk 1]
```diff
@@ -130,11 +131,14 @@ def check_branch(current_ref, desired_branch)
 
         @logger.debug("current_branch: #{current_branch} - desired_branch: #{desired_branch}")
 
+        local_branch = `git branch -a | grep -v "HEAD detached" | grep -r "#{desired_branch}"`.strip
```
Comment (hoatle) [inline lib/teracy-dev/location/git_synch.rb:134]: I don't think this is the proper way to get the local branch:

```
teracy-dev hoatle$ git branch -a | grep -v "HEAD detached" | grep -r "develop"
(standard input):* develop
(standard input):  remotes/datphan/develop
(standard input):  remotes/origin/HEAD -> origin/develop
(standard input):  remotes/origin/develop
(standard input):  remotes/upstream/develop
```

**Comment by hoatle** [comment]
we should handle gitsynch more properly, so I'll close this PR and the proper changes should be addressed on the issue https://github.com/teracyhq/dev/issues/465 and #469 instead

File: lib/teracy-dev/location/git_synch.rb (hunk 1 above)
Comment (datphan) [reply lib/teracy-dev/location/git_synch.rb:134]: It is a check for desired branch, not to get it, it is more correct than the current way,

**Pull request closed without merging**